### PR TITLE
Fix kwargs in JIT

### PR DESCRIPTION
**Pull request by Rohan138**
Minor bug fix: Currently calling the same `JIT`ted function with different kwargs breaks because the `expected_name_sts_dtype_device` changes.

This fixes the issue, but if it feels hacky, I can add a test or find a better solve (imo `itertools.chain`-ing the args and kwargs together isn't the best way of doing this).

### CODE REVIEW

**Comment by chenyuxyz** [comment]
can you include a regression test that fails on master?

**Comment by Rohan138** [comment]
Added test and fixed lint

File: test/test_jit.py [hunk 1]
```diff
@@ -96,6 +96,21 @@ def add_kwargs(first, second): return (first+second).realize()
       np.testing.assert_allclose(c.numpy(), a.numpy()+b.numpy(), atol=1e-4, rtol=1e-5)
     assert_jit_cache_len(add_kwargs, 1)
 
+  def test_reorder_kwargs_jit(self):
+    @TinyJit
+    def add_kwargs(first, second): return (first+second).realize()
```
Comment (geohot) [inline test/test_jit.py:101]: Can you make the test something non commutative so we can confirm the order is right?

Comment (Rohan138) [reply test/test_jit.py:101]: Actually, good catch. `first/second` breaks for example. Let me see if there's a better fix.

**Comment by Rohan138** [comment]
Test is division now; switched to using `inspect.signature(fxn)`. I think this is cleaner than the `enumerate(args)` + `sorted(kwargs)`.

Fixing `export_model` errors now, but let me know if you want me to rollback to the current way of doing it instead.

**Comment by Rohan138** [comment]
NVM; keeping it as is. I might switch to inspect.signature and clean up model export code (It's somewhat hacky rn) in a separate PR.

File: tinygrad/features/jit.py [hunk 2]
```diff
@@ -91,7 +91,7 @@ def __get__(self, obj, objtype): return functools.partial(self.__call__, obj)
 
   def __call__(self, *args, **kwargs) -> ReturnType:
     # all inputs (except const) are realized
-    input_tensors: Dict[Union[int, str], Union[LazyBuffer, MultiLazyBuffer]] = { cast(Union[int, str], k):v.realize().lazydata for k,v in itertools.chain(enumerate(args), kwargs.items()) if v.__class__ is Tensor }  # noqa: E501
+    input_tensors: Dict[Union[int, str], Union[LazyBuffer, MultiLazyBuffer]] = { cast(Union[int, str], k):v.realize().lazydata for k,v in itertools.chain(enumerate(args), sorted(kwargs.items(), key=lambda x: str(x[0]))) if v.__class__ is Tensor }  # noqa: E501
```
Comment (chenyuxyz) [inline tinygrad/features/jit.py:94]: Do you need to specify this `key=` function? I think it's the same default behavior without specifying the `key`

Comment (Rohan138) [reply tinygrad/features/jit.py:94]: No, `key` and `reverse` are kwarg-only parameters to sorted:
https://stackoverflow.com/questions/75761879/python-sorted-expected-1-argument-got-3-trying-to-understand-named-argument

Comment (chenyuxyz) [reply tinygrad/features/jit.py:94]: i mean just `sorted(kwargs.items())`

**Pull request merged**